pkg/client/log.go

```diff
@@ -4,25 +4,29 @@
 package client
 
 import (
-	"strings"
-
-	"github.com/sirupsen/logrus"
+	"go.uber.org/zap"
 )
 
-func GetLogger(format string, debug bool) *logrus.Logger {
-	logger := logrus.New()
-	logger.SetReportCaller(true)
+func GetLogger(format string, debug bool) *zap.Logger {
+	var logLevel zap.AtomicLevel
+	if debug {
+		logLevel = zap.NewAtomicLevelAt(zap.DebugLevel)
+	} else {
+		logLevel = zap.NewAtomicLevelAt(zap.InfoLevel)
+	}
 
-	switch strings.ToLower(format) {
-	case "text":
-		logger.SetFormatter(&logrus.TextFormatter{})
-	default:
-		logger.SetFormatter(&logrus.JSONFormatter{})
+	cfg := zap.Config{
+		Encoding: format,
+		Level:    logLevel,
 	}
 
-	if debug {
-		logger.SetLevel(logrus.DebugLevel)
+	logger, err := cfg.Build()
+	if err != nil {
+		panic("Failed to initialize logger: " + err.Error())
 	}
+	defer logger.Sync()
```

Comment:
should we be letting the caller be responsible for deferring this `logger.Sync()` call?

Reply:
(linty seems to be sad that we're not checking the err, can prob add a `// nolint:...`